File: BE/hieuthuoc/src/main/java/com/example/hieuthuoc/entity/PhieuNhap.java
package com.example.hieuthuoc.entity;

import java.util.List;

import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@Entity
@Table(name = "phieu_nhap")
@EntityListeners(AuditingEntityListener.class)
@EqualsAndHashCode(callSuper = true)
public class PhieuNhap extends TimeAuditable{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "nha_cung_cap_id")
    private NhaCungCap nhaCungCap;

    @ManyToOne
    @JoinColumn(name = "nguoi_dung_id")
    private NguoiDung nguoiDung;

    private Double tongTien;
    
    @OneToMany(mappedBy = "phieuNhap",fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ChiTietPhieuNhap> chiTietPhieuNhaps;

    // getters and setters
}

File: front/src/app/_model/phieunhap.ts
import { NguoiDung } from "./auth/nguoidung";
import { ChiTietPhieuNhap } from "./chitietphieunhap";
import { NhaCungCap } from "./ncc";

export class PhieuNhap {
  id?: string;
  ngayNhap?: string;
  tongTien?: string;
  nguoiDungId?: string;
  nguoiDung?: NguoiDung;
  nhaCungCapId?: string;
  nhaCungCap?: NhaCungCap;
  createdAt?: string;
  updatedAt?: string;
  hanSuDung?: string;

  chiTietPhieuNhaps?: ChiTietPhieuNhap[];
}
